src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -107,6 +177,7 @@ module.exports = function(scheduleUpdate : (fiber: Fiber, priorityLevel : Priori
     const ctor = workInProgress.type;
     const props = workInProgress.pendingProps;
     const instance = new ctor(props);
+    checkClassInstance(workInProgress, instance);
```

Comment:
Can we avoid checking if we are resuming work and recreating the instance? Check should only be done once.